### PR TITLE
Preallocate resulting array

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

File: src/Control/Apply.js [hunk 1]
```diff
@@ -2,11 +2,14 @@
 
 exports.arrayApply = function (fs) {
   return function (xs) {
-    var result = [];
+    var l = fs.length;
+    var k = xs.length;
+    var result = Array(l*k);
```
Comment (garyb) [inline src/Control/Apply.js]: `new Array`? I know it's not a requirement for `Array`, I just find it super weird to construct without `new`.

**Review by garyb — APPROVED** [review]
LGTM!

Did you benchmark it at all, any results to share?

**Comment by jacereda** [comment]
I only tested a 1000 floats array:

Before:
```
   apply/1000flt    32478246.483 ns 0.997 r2           30.790 ops/s 
```

After:
```
   apply/1000flt    16071140.637 ns 0.996 r2           62.223 ops/s 
```

**Comment by jacereda** [comment]
Well, to be clear, the measured thing was applying 1000 `id` to 1000 floats.

**Comment by garyb** [comment]
Okidoke, I just wondered. I expect it'd almost certainly be faster, but yknow... JavaScript 😆

**Review by michaelficarra — APPROVED** [review]

**Comment by paf31** [comment]
👍 Thanks!

**Comment by paf31** [comment]
@garyb Shall we merge this?

**Comment by garyb** [comment]
Sure!

**Pull request merged**